google/cloud/firestore_v1/base_document.py

```diff
@@ -503,6 +522,45 @@ def _to_protobuf(self) -> Optional[Document]:
         return _helpers.document_snapshot_to_protobuf(self)
 
 
+class QueryResultsList(list):
```

Comment:
maybe we should move this in with stream_generator.py? Or it's own file?

It made sense to leave this here when it only held DocumentReferences, but not so much now that it's generic

Reply:
True! moved

Reply:
I moved to its own file instead.